File: core/src/main/scala/diplomacy/variant/standard/rule/MovementOrderGenerator.scala
package diplomacy.variant.standard.rule

import diplomacy.rule.InvalidOrderMessage
import diplomacy.board.Power

class MovementOrderGenerator[Turn_ <: Turn, Power_ <: Power] extends OrderGenerator[Turn_, Power_] {
  def ordersToSkipPhase(board: Board): Option[Set[Order]] = {
    if (board.units.isEmpty) {
      Option(Set())
    } else {
      None
    }
  }
  def defaultOrder(board: Board)(unit: DiplomacyUnit): Order = Order.Hold(unit)
}

File: lib/variant/standard/rule/movement-order-generator.js
const { Hold } = require('./order')

module.exports = class MovementOrderGenerator {
  defaultOrderOf (board, unit) {
    return new Hold(unit)
  }

  ordersToSkipPhase (board) {
    if (board.units.size === 0) {
      return new Set()
    } else {
      return null
    }
  }
}
